### ISSUE TITLE
Initial team, trainings before Mars journey. 

### ISSUE THREAD

**Issue by shirishag75** [mars-sim/mars-sim#87]
Whether it be a government/state sanctioned venture or a privately funded one, don't think it will be easy to have a team of 14 people at once. At the most as per current technology maybe a crew of 4-5 with a complement of bots and probably some bots already on the ground already building/built crude structures on the ground should be there. 

The team will need to train for a year or more as a unit before setting off to mars.  Supportive roles of astronauts should and can be trained if it is known that people can survive and thrive on mars.

**Comment by mokun** [comment]
We created a class called `TrainingType` last year. But we haven't got to the point of making use of it.

> /**
>  * Mars Simulation Project
>  * TrainingType.java
>  * @version 3.1.0 2017-10-03
>  * @author Manny Kung
>  */
> 
> package org.mars_sim.msp.core.person;
> 
> import org.mars_sim.msp.core.Msg;
> 
> public enum TrainingType {
> 
> 	BIOETHICAL							(Msg.getString("TrainingType.bioethical")), //$NON-NLS-1$
> 	NOAA_NEEMO							(Msg.getString("TrainingType.NOAANEEMO")), //$NON-NLS-1$
> 	NASA_Desert_RATS					(Msg.getString("TrainingType.NASADesertRATS")), //$NON-NLS-1$
> 	SURVIVAL_TRAINING					(Msg.getString("TrainingType.survival")), //$NON-NLS-1$
> 	SCUBA_DIVING_MASTER					(Msg.getString("TrainingType.scubaDiving")), //$NON-NLS-1$
> 	FLIGHT_SAFETY						(Msg.getString("TrainingType.flightSafety")), //$NON-NLS-1$
> 	SEARCH_AND_RESCUE					(Msg.getString("TrainingType.searchAndRescue")), //$NON-NLS-1$
> 	MOUNTAINEERING_MASTER				(Msg.getString("TrainingType.mountaineering")), //$NON-NLS-1$
> 	AIRBORNE_AND_RANGER_TRAINING		(Msg.getString("TrainingType.airborneRanger")), //$NON-NLS-1$
> 	HAUGHTON_MARS_GEOLOGICAL			(Msg.getString("TrainingType.haughtonMarsGeological")), //$NON-NLS-1$
> 	HALO_JUMPMASTER						(Msg.getString("TrainingType.HALOJump")), //$NON-NLS-1$
> 	MISHAP_PREVENTION 					(Msg.getString("TrainingType.mishapPrevention")), //$NON-NLS-1$
> 	;
> 
> 	//MARS_500_C							(Msg.getString("TrainingType.mars500C")), //$NON-NLS-1$
> 	//ANALOG_ENVIRONMENT					(Msg.getString("TrainingType.analogEnvironment")), //$NON-NLS-1$
> 	//UNDERSEA_MISSION						(Msg.getString("TrainingType.underseaMission")), //$NON-NLS-1$
> 
> 	private String name;
> 
> 	/** hidden constructor. */
> 	private TrainingType(String name) {
> 		this.name = name;
> 	}
> 
> 	public final String getName() {
> 		return this.name;
> 	}
> 
> 	@Override
> 	public final String toString() {
> 		return getName();
> 	}
> }
>

**Comment by mokun** [comment]
@shirishag75 , 

You may have wondered how a person would end up being in one of the 7 *specialist roles* ?

Currently, a person's *job* **solely** affect this *role prospect score*. 

Let's make an assumption that any of those prior training experiences of a person would also affect this "role picking" thing as well.

Will put `TrainingType` into use and commit the code change soon.

**Comment by Urwumpe** [comment]
What I quite liked as system is the system how you create characters in the BattleTech RPG by essentially following a simplified earlier life with stations like early childhood, schooling (until age 16) and advanced training as adult. At every stage of your life, there are random events that further refine the abilities of your character. 

Maybe the training type could be the final summary of this more complex path for certain aspects of the simulation.

**Comment by mokun** [comment]
> At every stage of your life, there are random events that further refine the abilities of your character.

Yes, other than prior trainings, we can also formulate some sort of *life experiences* or *life events* for the settlers prior to their arrival on Mars.

**Comment by mokun** [comment]
Sure. We can definitely be more creative in giving unique quality and some sort of *individuality* to each settler with *events*. 

Shall we brain what categories of events can be stored and how they can be used in future for that person ?

**Comment by Urwumpe** [comment]
Well, I would say, one important category of event would be graduating from school or training, which could have some effect on the skills that he has. 
Another could be previous research projects. For example, some settler could have had a post-grad research project that was so unsuccessful that he decided to not publish it, maybe having effect on his personality to be less optimistic. 

Next, there could be events that motivate people to settle on Mars, severing ties to Earth. For example loss, but there could also be bad decisions on Earth, like criminal careers, massive fails at work, a stuck career, etc.

**Comment by mokun** [comment]
> one important category of event would be graduating from school or training, which could have some effect on the skills that he has.

Yes all the previous trainings a settler have now affect that person's skills.

See https://github.com/mars-sim/mars-sim/commit/ccf7c7e6fac1e9834f41ae1945d723d453a9e59e

> had a post-grad research project that was so unsuccessful that he decided to not publish it, maybe having effect on his personality to be less optimistic.

This is more difficult to implement. How a personality quirk would affect that person's decision is not as easy. 

I suppose we could still model a person's inclination in how to spend his time and do things to publish a paper and to connect with others or to be alone. But we have to model it one at a time. 

We may have to wait until we create an architecture for activities that impact every minor decisions, those that are more flexible and go beyond the confine of those Tasks and Missions.

> to settle on Mars, severing ties to Earth...  loss, but there could also be bad decisions on Earth, like criminal careers, massive fails at work, a stuck career, etc.

Yes there are definitely reasons upon which they end up choosing to go to Mars and settle there.  We could spell them out and link how each would impact their behaviors and their future.

**Comment by mokun** [comment]
@Urwumpe 
@bevans2000 

At what points will we expand the activities beyond the surface of Mars ? Where to start to allow for returning to Earth/moon and traveling between them ? 

If we bring in tourists, it will certainly make sense that we send them back home when their vacations are over :)

**Comment by bevans2000** [comment]
I guess my question is, what does it give us that we need? Seems we have a few other ideas if how to change just the Mars based simulation.

**Comment by mokun** [comment]
In terms of traveling between Earth, Moon and Mars, i feel it opens up what settlers can possibly do to further contribute to the entire Tri-system economy in ways that we haven't imagined before.

Trainers could be a new breed of career professionals. They may opt for a short-term assignment to set foot on Mars for a 6 month rotation and go back to the Moon or Earth for their next assignment.

I see this feature would add to the stories of mars-sim, making it a bit more engaging.

**Comment by bevans2000** [comment]
So replace the job and role history with a generic Career History. We add training events as well?
I like this.

**Comment by bevans2000** [comment]
Life history is fine. Works for me. I have added this to 3.11 milestone. Need to create a discussion for 3.11 development
